File: frontend/types/policy.ts
export interface Policy {
  id: string;
  code: string;
  name: string;
  description?: string;

  enabled: boolean;
  priority: number;

  sessionTimeout: number;
  idleTimeout: number;
  simultaneousUse: number;

  downloadRate: number;
  uploadRate: number;

  burstDownload?: number;
  burstUpload?: number;

  dailyQuota?: number;
  monthlyQuota?: number;
  totalQuota?: number;

  addressList?: string;
  vlanId?: number;
  ipPool?: string;

  expirationDate?: string | null;
  loginSchedule?: string;

  createdAt: string;
  updatedAt: string;
}

export interface PolicyListResponse {
  success: boolean;
  data: Policy[];
}

export interface PolicyFormData {
  code: string;
  name: string;
  description: string;

  enabled: boolean;
  priority: number;

  sessionTimeout: number;
  idleTimeout: number;
  simultaneousUse: number;

  downloadRate: number;
  uploadRate: number;

  burstDownload?: number;
  burstUpload?: number;

  dailyQuota?: number;
  monthlyQuota?: number;
  totalQuota?: number;

  addressList?: string;
  vlanId?: number;
  ipPool?: string;

  expirationDate?: string | null;
  loginSchedule?: string;
}

File: backend/Domain/NAP/Policy.cs
namespace BDIP.Domain.NAP;

public sealed class Policy
{
    public Guid Id { get; init; } = Guid.NewGuid();

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    // Session
    public int SessionTimeout { get; set; }
    public int IdleTimeout { get; set; }
    public int SimultaneousUse { get; set; } = 1;

    // Bandwidth
    public int DownloadRate { get; set; }
    public int UploadRate { get; set; }
    public int? BurstDownload { get; set; }
    public int? BurstUpload { get; set; }
    public int? Priority { get; set; }

    // Quota
    public long? DailyQuota { get; set; }
    public long? MonthlyQuota { get; set; }
    public long? TotalQuota { get; set; }

    // Network
    public string? AddressList { get; set; }
    public int? VlanId { get; set; }
    public string? IpPool { get; set; }

    // Access
    public bool Enabled { get; set; } = true;
    public DateTime? ExpirationDate { get; set; }
    public string? LoginSchedule { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}